feat: 월간 일정 UI 구현

Jihye817 opened the pull request:
## 💡 작업 내용

- [x] react-big-calendar, date-fns 라이브러리 추가
- [x] 월간 일정 UI 구현
- [x] 달력 css 변경
- [x] css 스타일을 변수로 사용할 수 있도록 변경

## 💡 자세한 설명

### 1️⃣ 라이브러리 추가
달력 구현을 위해 `react-big-calendar` 와 `date-fns` 라이브러리를 추가했습니다.
타입스크립트 사용을 위해 `@types/react-big-calendar` 라이브러리도 같이 추가했습니다!

### 2️⃣ 월간 일정 UI 구현
mockData를 이용해 일정이 표시되는 것을 확인 완료하였습니다. 현재 drag-drop 등의 기능은 적용되어 있지 않으며, 추후 필요 시 적용 예정입니다.
![Screenshot 2025-04-06 at 5 01 07 pm](https://github.com/user-attachments/assets/f7c5ee8d-149b-4bef-92b6-286a66cc5036)

### 3️⃣ 달력 css 변경
css 오버라이딩을 통해 달력의 css를 변경하기 위해 `calendar` 폴더 내부에 `calendar.css` 파일을 작성하였습니다. import를 통해 다른 파일에서 사용 가능합니다.

### 4️⃣ globals.css 변경 : css 변수로 사용하기
달력 css 파일에서 같은 색상을 사용하기 위해 기존의 `@theme inline` 부분에 들어 있던 색상 토큰 정의부분을 `:root` 부분으로 변경하였습니다. 다른 css 파일에서 변수 형태로 사용할 수 있습니다.

#### 사용예시
```css
.rbc-month-view {
    border: 1px solid var(--primary-100);
    border-radius: 10px 10px 0px 0px;
}
```

### *️⃣ 기타 사항
- globals.css에 일정 표시에 필요한 색상들을 추가하였습니다.
- 일간 UI 페이지의 오타 수정하였습니다.

## 📗 참고 자료 (선택)

## 📢 리뷰 요구 사항 (선택)

## ✅ 셀프 체크리스트

- [x] 머지할 브랜치 확인했나요?
- [x] Reviewers, Labels, Projects를 등록했나요?
- [x] 기능이 잘 동작하나요?
- [x] 불필요한 코드는 제거했나요?

Jihye817 committed:
TTOL-11 style: 체크박스 색상 및 완료된 일정 배경 색상  변경

Jihye817 committed:
TTOL-11 feat: BoardTitle 컴포넌트 생성

보드 제목을 재사용할 수 있도록 컴포넌트로 분리

Jihye817 committed:
TTOL-11 feat: 일간 일정 UI 구현

Jihye817 committed:
TTOL-11 style: 일정 완료 시 장소 아이콘 색상 수정

Jihye817 committed:
TTOL-11 style: 화면 너비에 맞춰 div의 너비가 변경되도록 수정

Jihye817 committed:
TTOL-11 style: 화면 너비에 맞춰 div의 너비가 변경되도록 수정

Jihye817 committed:
TTOL-13 chore: react-big-calendar 및 date-fns 라이브러리 추가

Jihye817 committed:
TTOL-13 style: css 스타일을 변수로 사용할 수 있도록 변경

:root 에 css 스타일 추가

Jihye817 committed:
TTOL-13 style: css 색상 추가 및 달력 css 변경

Jihye817 committed:
TTOL-13 feat: 월간 일정 UI 구현

월간 일정 UI 구현, 일간 일정 오타 수정

Jihye817 committed:
TTOL-11 refactor: children을 옵셔널로 지정

Jihye817 committed:
TTOL-11 refactor: 오타 수정, Task 타입 인터페이스로 정의하여 사용

rkdcodus commented on src/app/(afterlogin)/calendar/monthly/page.tsx:
```diff
@@ -0,0 +1,134 @@
+    },
+  ]);
+  const [finishedTaskCount, setFinishedTaskCount] = useState(0);
+  const [calendarMockData] = useState(
+    mockData.map(task => ({
+      ...task,
+      start_time: new Date(task.start_time),
+      end_time: new Date(task.end_time || task.start_time),
+    })),
+  );
```
캘릭터 데이터로 넣으려면 start_time, end_time이 Date 타입이여야하는군요!

Date로 변환해주는 작업이 계속 반복되지 않도록 하는게 중요할 것 같습니다 
어차피 서버에서 오는 데이터라 useState로 관리되지 않을 수 있는데 useMemo나 useEffect로 서버에서 넘어오는 값이 변경될 때만 Date로 변환되도록 해주는 처리해주면 좋을 것 같아요!

Jihye817 replied:
좋은 의견인 것 같습니다! useEffect 사용해서 mockData가 변경될 때 해당 로직이 실행되도록 구현해둘게요!

rkdcodus commented on src/app/(afterlogin)/calendar/monthly/page.tsx:
```diff
@@ -0,0 +1,134 @@
+      end_time: new Date(task.end_time || task.start_time),
+    })),
+  );
+  const locales = { ko };
+
+  const localizer = dateFnsLocalizer({
+    format,
+    parse,
+    startOfWeek,
+    getDay,
+    locales,
+  });
```
```suggestion
  const localizer = dateFnsLocalizer({
    format,
    parse,
    startOfWeek,
    getDay,
    locales = { ko },
  });
```
간단하게 작성해도 좋을 것 같습니다! 참고 사항입니다!

rkdcodus reviewed (approved):
주간 달력 구현하려고 보니 지혜님께서 작성해두신 색상 토큰 사용이나 페이지 부가 요소들 합치려면 머지하고 구현하는게 좋을 것 같아 먼저 리뷰했습니다!
전반적으로 큰 문제 없는 거 같아요!!  수고하셨습니다 👍

Jihye817 committed:
TTOL-13 refactor: string -> Date 변환 로직 리팩토링

useEffect를 사용하여 데이터 변환 시에만 Date로 변환하도록 변경 및 이에 따른 Task 타입 변경, localizer 변수 간단히 작성

rkdcodus commented on src/app/_types/index.ts:
```diff
@@ -0,0 +1,11 @@
+export interface Task {
+  task_id: number;
+  title: string;
+  memo: string;
+  start_time: string | Date;
+  end_time: string | Date;
+  address: string;
+  place_name: string;
+  location: { lat: string; lng: string };
+  is_completed: boolean;
+}
```
아아 Date 타입이 추가되어야하네요..!
start_time, end_time이 서버로 보내는 값은 string(ISO 8601 형식)이여서 개발하다가 변환시키는걸 놓치는 경우가 생길 수도 있을 거 같아요
Base 타입 하나 만들어두고 확장해서 하나는 TaskPayload , 하나는 TaskClient (타입명은 수정 가능) 이런식으로 분리시키는 건 어떨까요?!

Jihye817 replied:
좋아요!! Task가 Date 형식으로 쓰여야 하는 부분이 달력이 있을 때만 사용되고 있으니 TaskPayload와 TaskCalendar로 나누어 두겠습니다!

Jihye817 committed:
TTOL-13 refactor: Task 타입 확장

rkdcodus reviewed (approved):
넵! 확인했습니다!! 수고하셨습니다 🥰
충돌 부분만 해결하면 될 것 같아요

Jihye817 committed:
Merge branch 'dev' into TTOL-13-feature-monthly-tasks-ui
